Mathlib/NumberTheory/NumberField/Units.lean

```diff
@@ -47,10 +51,118 @@ attribute [local instance] NumberField.ringOfIntegersAlgebra
 
 variable {K}
 
-theorem isUnit_iff_norm [NumberField K] (x : 𝓞 K) :
+theorem isUnit_iff_norm [NumberField K] {x : 𝓞 K} :
     IsUnit x ↔ |(RingOfIntegers.norm ℚ x : ℚ)| = 1 := by
   convert (RingOfIntegers.isUnit_norm ℚ (F := K)).symm
   rw [← abs_one, abs_eq_abs, ← Rat.RingOfIntegers.isUnit_iff]
 #align is_unit_iff_norm isUnit_iff_norm
 
 end IsUnit
+
+namespace NumberField.Units
+
+section coe
+
+/-- The `MonoidHom` from the group of units `(𝓞 K)ˣ` to the field `K`. -/
+def coe_to_field : (𝓞 K)ˣ →* K := (Units.coeHom K).comp (map (algebraMap (𝓞 K) K))
+
+variable {K}
+
+/-- The coercion of `x : (𝓞 K)ˣ` into `K`. -/
+@[coe] def to_field (x : (𝓞 K)ˣ) : K := coe_to_field K x
+
+variable (K)
+
+theorem coe_to_field_injective : Function.Injective (coe_to_field K) :=
+  fun _ _ h => Units.eq_iff.mp (SetCoe.ext h)
+
+/-- There is a natural coercion from `(𝓞 K)ˣ` to `(𝓞 K)` and then from `(𝓞 K)` to `K` but it is
+useful to also have a direct one from `(𝓞 K)ˣ` to `K`. -/
+instance : Coe (𝓞 K)ˣ K := ⟨to_field⟩
```

Comment:
Could you explain in a bit more detail why this is useful? If I remove the instance Lean will take the existing `(𝓞 K)ˣ → (𝓞 K) → K` route in its coercions, and the file still builds (and `simp` can prove most of these results apart from `map_zpow`).

Reply:
You are right. I guess the coercion is not needed anymore with Lean4. I have removed it but kept some lemmas that I think are useful (with name changes): 
```lean
theorem coe_injective : Function.Injective ((↑) : (𝓞 K)ˣ → K)
theorem coe_pow (x : (𝓞 K)ˣ) (n : ℕ) : (x ^ n : K) = (x : K) ^ n
theorem coe_zpow (x : (𝓞 K)ˣ) (n : ℤ) : (x ^ n : K) = (x : K) ^ n
theorem coe_one : ((1 : (𝓞 K)ˣ) : K) = (1 : K)
theorem coe_ne_zero (x : (𝓞 K)ˣ) : (x : K) ≠ 0
```

Reply:
I agree with Riccardo that either the `rfl`s (including `coe_one`) should all be removed or all be kept.